Support git insteadOf config entries during import

smith-doug opened the pull request:
ls-remote --get-url will apply any special insteadOf rules.  This prevents false "Path already exists and contains a different repository" errors if you have something like the below in your gitconfig. 
```
[url "git@github.com:"]
    insteadOf = https://github.com/
```

smith-doug committed:
Use ls-remote --get-url

The works with git config insteadOf entries

smith-doug committed:
Output an info message when command.url is modified

christophebedard commented:
Would probably be good and easy enough to add a test here: https://github.com/dirk-thomas/vcstool/blob/master/test/test_commands.py

smith-doug committed:
Replace f strings

Not available in 3.5

smith-doug commented:
Ty for reminding me about tests.  The extra output I added broke them.  Will update.

smith-doug committed:
Remove print in git.py

Additional output breaks tests

smith-doug committed:
Add optional env override and raw_output options to run_command

smith-doug committed:
Add test for import when ssh->https insteadOf import rule exists

Git doesn't make it easy to override the location of the global gitconfig file.  Maybe I should just write the .gitconfig directly, to avoid any chance over overwriting the user's gitconfig

https->ssh would be more annoying as then you have to deal with ssh keys, making sure they exist, unlocking, etc.

smith-doug committed:
Write a local .gitconfig manually

Avoid any potential shell issues that might result in the user's real .gitconfig being touched.
